plugin/plugin.go

```diff
@@ -21,47 +18,29 @@ import (
 	"github.com/weaveworks/weave/plugin/skel"
 )
 
-var version = "unreleased"
-
 var Log = common.Log
 
-func main() {
-	var (
-		justVersion      bool
-		address          string
-		meshAddress      string
-		logLevel         string
-		noMulticastRoute bool
-	)
-
-	flag.BoolVar(&justVersion, "version", false, "print version and exit")
-	flag.StringVar(&logLevel, "log-level", "info", "logging level (debug, info, warning, error)")
-	flag.StringVar(&address, "socket", "/run/docker/plugins/weave.sock", "socket on which to listen")
-	flag.StringVar(&meshAddress, "meshsocket", "/run/docker/plugins/weavemesh.sock", "socket on which to listen in mesh mode")
-	flag.BoolVar(&noMulticastRoute, "no-multicast-route", false, "deprecated (this is now the default)")
-
-	flag.Parse()
-
-	if justVersion {
-		fmt.Printf("weave plugin %s\n", version)
-		os.Exit(0)
-	}
-
-	common.SetLogLevel(logLevel)
+func Start(weaveAPIAddr string, dockerAPIAddr string, address string, meshAddress string) {
+	weave := weaveapi.NewClient(weaveAPIAddr, Log)
 
-	weave := weaveapi.NewClient(os.Getenv("WEAVE_HTTP_ADDR"), Log)
-
-	// API 1.21 is the first version that supports docker network commands
-	dockerClient, err := docker.NewVersionedClientFromEnv("1.21")
-	if err != nil {
-		Log.Fatalf("unable to connect to docker: %s", err)
+	var dockerClient *docker.Client
+	var err error
+	if dockerAPIAddr != "" {
+		// API 1.21 is the first version that supports docker network commands
+		dockerClient, err = docker.NewVersionedClient(dockerAPIAddr, "1.21")
```

Comment:
Due to different versions. Anyway, I made it re-use the one from weaver which uses unversioned instance of Docker client. After inspecting `go-dockerclient`, I don't see any point of pinning versions (besides min-api-vsn check in dockerd which is not relevant to us).